persistence/cosmosdb/transactions_information_CosmosDB_[1.1,].partial.md

```diff
@@ -0,0 +1,91 @@
+A configuration API has been introduced as of version 1.1 and above. This configuration API provides the ability to specify the partition and Container used per message using message headers or the message contents.
+
+Using message headers only has the advantage being able to identify the partition or `Container` before the [outbox](/nservicebus/outbox) logic is executed, which allows the outbox feature to work entirely as intended.
+
+In the case where the partition or Container cannot be identified using only the message headers, the message contents can be used. This works because the Cosmos DB persistence introduces additional outbox logic to locate the outbox record and bypass the remaining message processing pipeline at a later stage of processing.
+
+## Specifying the `Container` partition to use for the transaction
+
+Transactions can only be executed within a single `Container` partition. The `PartitionKey` must be specified for transactions to work in Azure Cosmos DB.
+
+### Using message header values
+
+A single message header value can be used to specify the `PartitionKey` for the partition:
+
+snippet: ExtractPartitionKeyFromHeaderSimple
+
+Multiple message header values can also be used. Additionally overloads exist that allow a state object to be provided and passed when the extractor is called to avoid unnecessary allocations:
+
+snippet: ExtractPartitionKeyFromHeadersExtractor
+
+A custom class that implements `IPartitionKeyFromHeadersExtractor` can be implemented to specify the `PartitionKey` using message headers:
+
+snippet: CustomPartitionKeyFromHeadersExtractor
+
+The `IPartitionKeyFromHeadersExtractor` implementation can be registered via dependency injection or configured via the API:
```

Comment:
Do we need to show how to register those?

Reply:
Added